nutils/log.py

```diff
@@ -487,27 +478,44 @@ def range( title, *args ):
   '''Progress logger identical to built in range'''
 
   items = _range( *args )
-  return _logiter( title, _iter(items), length=len(items), useitem=True )
+  log = _getlog()
+  for item in items:
+    with log.context( '{} {} ({:.0f}%)'.format( title, item, (item+.5) * 100. / len(items) ) ):
+      yield item
 
 def iter( title, iterable, length=None ):
   '''Progress logger identical to built in iter'''
 
-  return _logiter( title, _iter(iterable), length=length or _len(iterable) )
+  if length is None:
+    length = _len(iterable)
+  log = _getlog()
+  index = 0
+  for item in iterable:
+    text = '{} {}'.format( title, index )
+    if length:
+      text += ' ({:.0f}%)'.format( (index+.5) * 100. / length )
+    with log.context( text ):
+      yield item
+    index += 1
```

Comment:
Instead of

    index = 0
    for item in iterable:
        ....
        index += 1

you could write

    for index, item in enumerate( iterable ):
        ...